tests/unit/package_managers/test_gomod.py

```diff
@@ -1485,3 +1506,55 @@ def test_get_repository_name(mock_git_repo: Any, input_url: str) -> None:
     resolved_url = _get_repository_name(RootedPath("/my-folder/cloned-repo"))
 
     assert resolved_url == expected_url
+
+
+@pytest.fixture
+def repo_remote_with_tag(rooted_tmp_path: RootedPath) -> tuple[RootedPath, RootedPath]:
+    """
+    Return the Paths to two Repos, with the first configured as the remote of the second.
+
+    There are different git tags applied to the first and second commits of the README file
+    """
+    local_repo_path = rooted_tmp_path.join_within_root("local")
+    remote_repo_path = rooted_tmp_path.join_within_root("remote")
+    readme_file_path = remote_repo_path.join_within_root("README.md")
+
+    local_repo_path.path.mkdir()
+    remote_repo_path.path.mkdir()
+    remote_repo = git.Repo.init(remote_repo_path)
+
+    with open(readme_file_path, "wb"):
+        pass
+    remote_repo.index.add([readme_file_path])
+    initial_commit = remote_repo.index.commit("Add README")
+
+    with open(readme_file_path, "w") as f:
+        f.write("This is a README")
+    remote_repo.index.add([readme_file_path])
+    remote_repo.index.commit("Update README")
+
+    git.Repo.clone_from(remote_repo_path, local_repo_path)
+
+    remote_repo.create_tag("v1.0.0", ref=initial_commit)
+    remote_repo.create_tag("v2.0.0")
+
+    return remote_repo_path, local_repo_path
+
+
+def test_fetch_tags(repo_remote_with_tag: tuple[RootedPath, RootedPath]) -> None:
+    _, local_repo_path = repo_remote_with_tag
+    version_resolver = ModuleVersionResolver(local_repo_path)
+    version_resolver.fetch_remote_tags()
+    assert version_resolver._commit_tags == ["v2.0.0"]
+    assert version_resolver._all_tags == ["v1.0.0", "v2.0.0"]
```

Comment:
~~Can we also check that both properties are `[]` before we fetch the tags?~~

I think we can't because the properties are cached. That makes my suggestion to split `fetch_remote_tags()` from `__init__` more questionable than I thought :sweat_smile: 

Maybe fetch_remote_tags could reset the cached properties, or it really shouldn't be possible to create a VersionResolver without fetching tags

Reply:
Added a classmethod. Let me know what you think :pray: